### PR TITLE
C# 8 phase 1

### PR DESCRIPTION
**Pull request by vslee**
- jjxtra began by marking some objects using the nullable (?) reference type signifier

### CODE REVIEW

File: ExchangeSharp/Utility/CryptoUtility.cs [hunk 1]
```diff
@@ -1075,37 +1075,39 @@ public static byte[] AesEncryption(byte[] input, byte[] password, byte[] salt)
...
+                AES.Key = key.GetBytes(AES.KeySize / 8);
+                AES.IV = key.GetBytes(AES.BlockSize / 8);
+                AES.Mode = CipherMode.CBC;
+                MemoryStream encrypted = new MemoryStream(input);
+                byte[] saltMatch = new byte[salt.Length];
+                if (encrypted.Read(saltMatch, 0, saltMatch.Length) != salt.Length || !salt.SequenceEqual(saltMatch))
+                {
+                    throw new InvalidOperationException("Invalid salt");
+                }
+                var cs = new CryptoStream(encrypted, AES.CreateDecryptor(), CryptoStreamMode.Read);
```
Comment (johnnyasantoss) [inline ExchangeSharp/Utility/CryptoUtility.cs]: Maybe here would be good to use `using var` expression.

Comment (vslee) [reply ExchangeSharp/Utility/CryptoUtility.cs]: Can you add an example of what you mean? Then I can add a commit to this PR. Alternatively, I can merge this PR and you can put it into your current PR or a new one.

Comment (johnnyasantoss) [reply ExchangeSharp/Utility/CryptoUtility.cs]: I was referring to the same expression in the second example of [this post](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement#example).
Also, I can add it later :)

**Comment by vslee** [comment]
Need @jjxtra to put an approving review before I can merge (I know it's strange for him to be reviewing his own code but since I was the one starting the PR, I can't review my own PR)

**Comment by jjxtra** [comment]
I don't know if I approve of my code to be honest :) We might consider backing out of nullable reference types on a project level and going to a per file level to reduce the warning barrage.

**Comment by vslee** [comment]
Far fewer warnings now (~50 instead of ~500)

**Comment by vslee** [comment]
I think we can merge now, and then whittle the rest of the warnings down with time.

**Comment by vslee** [comment]
progress towards #450

**Pull request merged**